arangod/Pregel/PregelMetrics.cpp

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "PregelMetrics.h"
+
+#include "Pregel/PregelMetricsDeclarations.h"
+#include "Metrics/MetricsFeature.h"
+
+using namespace arangodb::pregel;
+
+PregelMetrics::PregelMetrics(metrics::MetricsFeature& metricsFeature)
+    : ReplicatedLogMetrics(&metricsFeature) {}
+
+template<typename Builder, bool mock>
+auto PregelMetrics::createMetric(metrics::MetricsFeature* metricsFeature)
+    -> std::shared_ptr<typename Builder::MetricT> {
+  TRI_ASSERT((metricsFeature == nullptr) == mock);
+  if constexpr (!mock) {
+    return metricsFeature->addShared(Builder{});
+  } else {
+    return std::dynamic_pointer_cast<typename Builder::MetricT>(
+        Builder{}.build());
+  }
+}
+
+template<
+    typename MFP,
+    std::enable_if_t<std::is_same_v<arangodb::metrics::MetricsFeature*, MFP> ||
```

Comment:
no sure if we can use concepts now in ArangoDB: If yes, can't this enable_if be exchanged by a concept?